tools/cldr-code/src/main/java/org/unicode/cldr/test/ExampleGenerator.java

```diff
@@ -2861,6 +2888,87 @@ private String handleDateRangePattern(String value) {
         return result;
     }
 
+    /**
+     * Add examples for eras. First checks if there is info for this
+     *  calendar type and this era type in the CALENDAR_ERAS map, then
+     * generates a sample date based on this info and formats it
+     */
+    private String handleEras(XPathParts parts, String value) {
+        String calendarId = parts.getAttributeValue(3, "type");
+        String type = parts.getAttributeValue(-1, "type");
+        String id = (calendarId.contains("islamic")) ? "islamic" : calendarId; // islamic variations map to same sample
+        if (!CALENDAR_ERAS.containsKey(id)) {
+            return null; 
+        }
+        int typeIndex = Integer.parseInt(type);
+        if (calendarId.equals("japanese")) { 
+            if (typeIndex < 235) { // examples only for 2 most recent eras
+                return null; 
+            } else {
+                typeIndex %= 235; // map to length 2 list
+            }
+        }
+        List<Date> eraDates = CALENDAR_ERAS.get(id);
+        Date sample = eraDates.get(typeIndex);
+        String skeleton = "Gy";
+        String checkPath =
+                "//ldml/dates/calendars/calendar[@type=\""
+                        + calendarId
+                        + "\"]/dateTimeFormats/availableFormats/dateFormatItem[@id=\""
+                        + skeleton
+                        + "\"]";
+        String dateFormat = cldrFile.getWinningValue(checkPath);
```

Comment:
The parts value has the original path in it, so you should use that.

Reply:
I want to make sure I understand this correctly--the original path would be something like `"//ldml/dates/calendars/calendar[@type="gregorian"]/eras/eraNames/era[@type="0"]"`, so to get the path I want to use for the formatting I'd do a few replacements (ex. `origPath.replace("eras", "dateTimeFormats").replace("eraNames", "availableFormats").replace(...);`, and similarly for the comment below. I guess I wanted to clarify that the path I'd get from the parts variable isn't the path I'm using in these two cases (although it can be modified to be).

Reply:
You are right! I read this too quickly. Yes, you do want to use the patterns from availableFormats in the example, as well as the value from this path. Same for the other one.